extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -70,11 +71,46 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return data;
 	}
 
-	public serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Uint8Array {
-		return new TextEncoder().encode(this.serializeNotebookToString(data));
+	public async serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Promise<Uint8Array> {
+		return new TextEncoder().encode(await this.serializeNotebookToString(data));
 	}
 
-	public serializeNotebookToString(data: vscode.NotebookData): string {
+	private async serializeViaWorker(workerData: notebookSerializationWorkerData): Promise<string> {
+		const workerThreads = await import('node:worker_threads');
+		const path = await import('node:path');
+		const { Worker } = workerThreads;
+
+		return await new Promise((resolve, reject) => {
+			const workerFile = path.join(__dirname, 'notebookSerializerWorker.js');
+			const worker = new Worker(workerFile, { workerData });
+			worker.on('message', resolve);
+			worker.on('error', reject);
+			worker.on('exit', (code) => {
+				if (code !== 0) {
+					reject(new Error(`Worker stopped with exit code ${code}`));
+				}
+			});
+		});
+	}
+
+	private serializeNotebookToJSON(notebookContent: Partial<nbformat.INotebookContent>, indentAmount: string): Promise<string> {
+
+		const isInNodeJSContext = typeof process !== 'undefined' && process.release && process.release.name === 'node';
+		const experimentalSave = vscode.workspace.getConfiguration('ipynb').get('experimental.serialization', false);
+		if (isInNodeJSContext && experimentalSave) {
+			return this.serializeViaWorker({
+				notebookContent,
+				indentAmount
+			});
+		} else {
+			// ipynb always ends with a trailing new line (we add this so that SCMs do not show unnecessary changes, resulting from a missing trailing new line).
+			const sorted = sortObjectPropertiesRecursively(notebookContent);
+
+			return Promise.resolve(JSON.stringify(sorted, undefined, indentAmount));
```

Comment:
it seems we dont' append the additional `\n` anymore?

Reply:
Yes, good spot, thanks!